src/Simplify.elm

```diff
@@ -6834,6 +6834,24 @@ listGetElements resources =
                         Nothing ->
                             Just { known = [ head ], allKnown = False }
 
+                _ ->
+                    Nothing
+        , \expressionNode ->
+            case AstHelpers.removeParens expressionNode of
+                Node _ (Expression.OperatorApplication "++" _ leftList rightList) ->
+                    case ( listGetElements resources leftList, listGetElements resources rightList ) of
+                        ( Just leftElements, Just rightElements ) ->
+                            Just { allKnown = leftElements.allKnown && rightElements.allKnown, known = leftElements.known ++ rightElements.known }
+
+                        ( Just leftElements, Nothing ) ->
+                            Just { known = leftElements.known, allKnown = False }
+
+                        ( Nothing, Just rightElements ) ->
+                            Just { known = rightElements.known, allKnown = False }
+
+                        _ ->
```

Comment:
Since we already know this can only be the concrete case `( Nothing, Nothing )` I'd do an exhaustive match